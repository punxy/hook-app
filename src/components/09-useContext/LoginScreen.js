import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const {setUser} = useContext(UserContext);

    const user = {
        id: 123,
        name: 'Sebasti√°n Silva',
        email: 'sebasilvac88@gmail.com'
    }

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />

            <button
                className='btn btn-primary'
                onClick={() => { setUser(user) }}
            >setUser</button>
        </div>
    )
}
