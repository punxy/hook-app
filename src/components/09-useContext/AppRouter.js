import React from 'react';
import {
    BrowserRouter as Router,
    Routes,
    Route
} from "react-router-dom";
import { NavBar } from './NavBar';

import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />

                <div className='container-fluid'>
                    {/* Version 6 */}
                    <Routes>
                        <Route path="/" element={<HomeScreen />} />
                        <Route path="login" element={<LoginScreen />} />
                        <Route path="about" element={<AboutScreen />} />
                        <Route path='*' element={<HomeScreen />} />
                    </Routes>
                    {/* Version 5
                        <Switch>
                        <Route exact path='/' component={ HomeScreen }/>
                        <Route exact path='/about' component={ AboutScreen }/>
                        <Route exact path='/login' component={ LoginScreen }/>
                    </Switch> */}
                </div>
            </div>
        </Router>
    )
}
