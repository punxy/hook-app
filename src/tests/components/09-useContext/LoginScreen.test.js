import React from 'react';
import { mount } from "enzyme";
import '@testing-library/jest-dom';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Test in LoginScreen component', () => {

    const setUser = jest.fn();
    const wrapper = mount(
        <UserContext.Provider value={{
            setUser
            }} >
            <LoginScreen  />
        </UserContext.Provider>
    );


    test('should be render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be excecute setLogin', () => { 
        
        wrapper.find('button').simulate('click');
        expect( setUser ).toHaveBeenCalledWith({
            id: 123,
            name: 'Sebasti√°n Silva',
            email: 'sebasilvac88@gmail.com'
        });

    });
});