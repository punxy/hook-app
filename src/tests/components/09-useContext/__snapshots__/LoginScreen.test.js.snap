// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in LoginScreen component should be render correctly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    setUser
  </button>
</div>
`;
